barretenberg/cpp/src/barretenberg/commitment_schemes/gemini/gemini_impl.hpp

```diff
@@ -235,51 +226,31 @@ std::vector<typename GeminiProver_<Curve>::Polynomial> GeminiProver_<Curve>::com
 };
 
 /**
- * @brief Computes/aggragates d+1 Fold polynomials and their opening pairs (challenge, evaluation)
- *
- * @details This function assumes that, upon input, last d-1 entries in fold_polynomials are Fold_i.
- * The first two entries are assumed to be, respectively, the batched unshifted and batched to-be-shifted
- * polynomials F(X) = ∑ⱼ ρʲfⱼ(X) and G(X) = ∑ⱼ ρᵏ⁺ʲ gⱼ(X). This function completes the computation
- * of the first two Fold polynomials as F + G/r and F - G/r. It then evaluates each of the d+1
- * fold polynomials at, respectively, the points r, rₗ = r^{2ˡ} for l = 0, 1, ..., d-1.
+ * @brief Computes partially evaluated batched polynomials A₀₊(X) = F(X) + G(X)/r and A₀₋(X) = F(X) - G(X)/r
  *
- * @param mle_opening_point u = (u₀,...,uₘ₋₁) is the MLE opening point
- * @param fold_polynomials vector of polynomials whose first two elements are F(X) = ∑ⱼ ρʲfⱼ(X)
- * and G(X) = ∑ⱼ ρᵏ⁺ʲ gⱼ(X), and the next d-1 elements are Fold_i, i = 1, ..., d-1.
- * @param r_challenge univariate opening challenge
+ * @param batched_F F(X) = ∑ⱼ ρʲfⱼ(X)
+ * @param batched_G G(X) = ∑ⱼ ρᵏ⁺ʲ gⱼ(X)
+ * @param r_challenge
+ * @param batched_groups_to_be_concatenated
+ * @return {A₀₊(X), A₀₋(X)}
  */
 template <typename Curve>
-std::vector<typename GeminiProver_<Curve>::Claim> GeminiProver_<Curve>::compute_fold_polynomial_evaluations(
-    const size_t num_variables,
-    std::vector<Polynomial>&& fold_polynomials,
-    const Fr& r_challenge,
-    std::vector<Polynomial>&& batched_groups_to_be_concatenated)
+std::pair<typename GeminiProver_<Curve>::Polynomial, typename GeminiProver_<Curve>::Polynomial> GeminiProver_<
+    Curve>::compute_partially_evaluated_batch_polynomials(const size_t log_n,
+                                                          Polynomial&& batched_F,
+                                                          Polynomial&& batched_G,
+                                                          const Fr& r_challenge,
+                                                          std::vector<Polynomial> batched_groups_to_be_concatenated)
 {
-
-    Polynomial& batched_F = fold_polynomials[0]; // F(X) = ∑ⱼ ρʲ   fⱼ(X)
-
-    Polynomial& batched_G = fold_polynomials[1]; // G(X) = ∑ⱼ ρᵏ⁺ʲ gⱼ(X)
-
-    // Compute univariate opening queries rₗ = r^{2ˡ} for l = 0, 1, ..., m-1
-    std::vector<Fr> r_squares = gemini::powers_of_evaluation_challenge(r_challenge, num_variables);
+    Polynomial& A_0_pos = batched_F; // A₀₊ = F
```

Comment:
The updated version of this calculation still only allocates 1 additional "scratch" polynomial but is much easier to read